llvm/lib/AsmParser/LLParser.cpp

```diff
@@ -8222,7 +8222,7 @@ int LLParser::parseAtomicRMW(Instruction *&Inst, PerFunctionState &PFS) {
               " operand must be an integer, floating point, or pointer type");
     }
   } else if (IsFP) {
-    if (!Val->getType()->isFloatingPointTy()) {
+    if (!Val->getType()->isFPOrFPVectorTy()) {
```

Comment:
What happens if you pass a scalable vector type? Should we prohibit that?

Reply:
Do any scalable architectures support vector atomics? I assume it would fail in codegen somewhere. Is that worse than having it be considered invalid IR?

Reply:
Thanks. ISTM there is some difference between "not all targets handle" and "the target-independent code cannot handle, and probably never will".